pkg/registry/extensions/daemonset/strategy_test.go

```diff
@@ -31,3 +37,67 @@ func TestDefaultGarbageCollectionPolicy(t *testing.T) {
 		t.Errorf("DefaultGarbageCollectionPolicy() = %#v, want %#v", got, want)
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		method                 string
+		url                    string
+		oldSelectorLabels      map[string]string
+		newSelectorLabels      map[string]string
+		expectedAPIVersion     string
+		expectedSelectorLabels map[string]string
+	}{
+		{"GET", "/api/v1beta2/namespaces", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "v1beta2", map[string]string{"a": "b"}},
+		{"GET", "/api/v1beta1/namespaces", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "v1beta1", map[string]string{"c": "d"}},
+	}
+
+	resolver := newTestRequestInfoResolver()
+
+	for _, test := range tests {
+		req, _ := http.NewRequest(test.method, test.url, nil)
+
+		apiRequestInfo, err := resolver.NewRequestInfo(req)
+		if err != nil {
+			t.Errorf("Unexpected error for url: %s %v", test.url, err)
+		}
+		if !apiRequestInfo.IsResourceRequest {
```

Comment:
These controllers never existed in the groupless `/api/` prefix, so your test shouldn't expect those inputs to work. Specifically, you want to make sure that `/apis/extensions/v1beta1` allows mutation, and `/apis/apps/v1beta2` does not. If we decide to plan ahead for `/apis/apps/v1` you could test for that too, but I don't know if that's decided yet.